src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module contains the ticking function that is responsible for invoking block acquisition functionality and executing it in periodic fashion.
+
+
+module Cardano.Wallet.BlockSyncer
+  (
+    blockSyncerBracket
+  ) where
+
+
+import Prelude
+
+import Cardano.Wallet.Primitive
+    ( Block (..) )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Exception
+    ( SomeException, bracket, catch )
+import Data.Text.Internal
+    ( showText )
+import Formatting
+    ( sformat, (%) )
+
+import qualified Formatting as F
+
+
+blockSyncerBracket
+    :: IO [Block]
+    -- ^ a way to get a new block
+    -> (Block -> IO ())
+    -- ^ action taken on a new block
+    -> Int
+    -- ^ tick time in seconds
+    -> IO ()
+blockSyncerBracket getNextBlocks action tickTime =
+    bracket
+      (return ())
+      (\_->return ())
+      (tickingFunction getNextBlocks action tickTime)
+
+tickingFunction
+    :: IO [Block]
+    -> (Block -> IO ())
+    -> Int
+    -> ()
+    -> IO ()
+tickingFunction getNextBlocks action tickTime _ =
+    go `catch`
+    (\(e :: SomeException) ->
+            let msg = "Terminating tickingFunction of BlockSyncer due to " % F.shown
+            in (putStrLn . showText) $ sformat msg e
+    )
```

Comment:
That is rather problematic to handle the error here in such way.. If the ticking function dies... as for now, we probably wants the wallet to die too. That's a fatal error and here, we'll just be showing a message in the console and silently fade away. The wallet server will keep running and... nothing will happen until we eventually notice that there's no block getting fetched anymore. 

So, until further needs, let just _do nothing_ and let the exception bubbles to the caller.

Reply:
removed and also formatting dependency